### DISCUSSION TITLE
org-ql-block in agenda doesn't return any results

### DISCUSSION THREAD

**Discussion by max6166** [alphapapa/org-ql#390] [Q&A]
I've been trying to get something very similar to the [org-ql-block](https://github.com/alphapapa/org-ql#function-org-ql-block) example in the docs to work. 

The agenda created with regular search terms populates correctly.
But the `org-ql-block` version doesn't return any results. It only displays the header string and is otherwise empty.

After 2 days of trying to figure out the problem, I'm still stumped. Any help would be greatly appreciated. 

```
(setq org-agenda-custom-commands
  '(("v" "org-ql: home 2DAY"
      ((org-ql-block '(and (todo "2DAY")
                        (tags "home"))
         ((org-ql-block-header "org-ql: 2DAY :home:")))))
     ("V" "search: home 2DAY"
       ((tags-todo "+home/!2DAY")))))
```

Here is an example org file:
```
* 2DAY Do things   :home: 
SCHEDULED: <2023-12-31 Sun>
```

EDIT: Added org-file example

**Comment by alphapapa** [comment]
You need to show an example Org file that you are searching in.

**Reply by max6166** [reply]
I added an example org file to the original question.

**Reply by alphapapa** [reply]
Have you configured Org to include `2DAY` as a to-do keyword?

**Reply by max6166** [reply]
Yes, 2DAY is a TODO keyword and :home: is a tag. I've been using them both for years.

As I mentioned, the normal custom agenda search in my example works fine. 

The terms also work in my custom agendas which use org-super-agenda.

No messages or warnings are generated when I run the org-ql agenda. I have debug-on-error turned on also.

**Reply by alphapapa** [reply]
In a clean Emacs config by running `with-emacs.sh -i org-ql`:

1.  Write to file `/tmp/test.org`:
```
#+TODO: TODO 2DAY | DONE

* 2DAY Do things   :home: 
SCHEDULED: <2023-12-31 Sun>
```
2.  `C-c C-c` on first line to refresh setup.
3. `C-c [` to add file to `org-agenda-files`.
4. Evaluate:
```el
(setq org-agenda-custom-commands
  '(("v" "org-ql: home 2DAY"
      ((org-ql-block '(and (todo "2DAY")
                        (tags "home"))
         ((org-ql-block-header "org-ql: 2DAY :home:")))))
     ("V" "search: home 2DAY"
       ((tags-todo "+home/!2DAY")))))
```
5. `M-x org-agenda RET v RET`.
6. Result: 
![Screenshot_20231231_173811](https://github.com/alphapapa/org-ql/assets/601365/36acea46-2ce7-47b6-b7e3-2df31e014a96)

So, it works for me.
